src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/Cas2ApplicationNoteEntity.kt

```diff
@@ -31,5 +30,29 @@ data class Cas2ApplicationNoteEntity(
 
   var body: String,
 ) {
+
+  @ManyToOne
+  @JoinColumn(name = "created_by_nomis_user_id")
+  var createdByNomisUser: NomisUserEntity? = null
+
+  @ManyToOne
+  @JoinColumn(name = "created_by_external_user_id")
+  var createdByExternalUser: ExternalUserEntity? = null
+
+  init {
+    when (this.createdByUser) {
+      is NomisUserEntity -> this.createdByNomisUser = this.createdByUser
+      is ExternalUserEntity -> this.createdByExternalUser = this.createdByUser
+    }
+  }
+
+  fun getCreatedByNomisOrExternalUser(): Cas2User {
```

Comment:
Have you considered calling this getCas2User() instead?

Reply:
good shout

Reply:
I've gone with `getUser` as per @tahb's [suggestion below](https://github.com/ministryofjustice/hmpps-approved-premises-api/pull/1500#discussion_r1494710694) - that perhaps those calling it don't need to know underlying implementation?

I originally wanted to call it `getCreatedByUser` because I'd prefer to be specific but got a Java(or Hibernate?) complaint that it clashed with the generated `get` method for the field `createdByUser` - I tried to look into best practices/solutions for overriding the generated one, and wasn't sure what to do, if you have any ideas?

Reply:
That makes sense. This is where my knowledge of kotlin is limited - I always get confused around constructors! I think there's a way to declare that the 'createdByUser' is _not_ a property and a constructor arg only (drop the 'val' keyword?), but i'm not sure if it would then be available to init. Maybe give that a quick go and if it doesn't work go with getUser() like suggested 👍